packages/betterer/src/reader.ts

```diff
@@ -1,20 +1,27 @@
-import { readFile } from 'fs';
-import { promisify } from 'util';
-
+import { existsSync } from 'fs';
 import { BettererResults } from './types';
-
-const readFileAsync = promisify(readFile);
+import * as dedent from 'dedent';
+import to from 'await-to-js';
 
 export async function read(resultsPath: string): Promise<BettererResults> {
-  try {
-    await readFileAsync(resultsPath);
-  } catch {
+  if (!existsSync(resultsPath)) {
     return {};
   }
 
-  try {
-    return await import(resultsPath);
-  } catch {
-    throw new Error();
+  const [importError, config = {}] = await to(import(resultsPath));
```

Comment:
I was assuming that if the import fails then that means something happened and the config got messed up somehow. I.e. it can't be read. But if it doesn't exists, that's fine we just return `{}`. 

If we just return `{}` in the catch statement when the import fails, that'll mean that if the file fails to be read (for whatever reason) then we'll just end up overwriting it. 

Is that the behavior you want?

Reply:
Concerning the `async-to-js` code...

You're right, it's not idiomatic. That said, it sort of forces you to do something with the error state in a way that try/catch doesn't. 

You're always presented with an error state first. You also have the option of handling what happens if say... there's an error _or_ the result is empty. Try/catch requires that handling those different states has to be done in different code paths. 

This is the basic usage.

```
const [error, result] = await to(doSomethingAsync());
```

I use it heavily as it provides an much more ergonomic mechanism for dealing  with async code, especially when you have to make decisions on multiple error states. 

Here's a good example from one of my projects:

https://github.com/auto-it/autobot/blob/2e526d03c86e40155cb89c48691c0dbae1d4ade4/lib/features/pr-onboarding.ts#L316-L337

Obviously it's not required and if you feel strongly about it I can definitely take it out. I'd encourage you to play around with it though. It really does make async/await code simpler.